client/src/main/scala/drt/client/components/TerminalPlanningComponent.scala

```diff
@@ -108,21 +108,25 @@ object TerminalPlanningComponent {
               <.th(^.colSpan := 2, ^.className := "heading", "Time"), sortedDays.flatMap(_ => List(<.th(^.className := "sub-heading", "Avail"), <.th(^.className := "sub-heading", "Rec"))).toTagMod
             )),
           <.tbody(
-            byTimeSlot.map(row => {
-              <.tr(
-                <.td(^.colSpan := 2, SDate(MilliDate(row.head.startMillis)).toHoursAndMinutes()),
-                row.flatMap(col => {
-                  val ragClass = TerminalDesksAndQueuesRow.ragStatus(col.required, col.available)
-                  List(<.td(^.className := ragClass, col.available), <.td(col.required))
-                }).toTagMod
-              )
-            }).toTagMod
+            byTimeSlot.zip(Forecast.timeSlotStartTimes(props.forecastPeriod.forecast, (millis: MillisSinceEpoch) => SDate(millis).toHoursAndMinutes())).map {
```

Comment:
maybe extract the Forecast.timeSlotStartTimes call to a val to make this line easier to read